global/segment-with-location@1.0.js

```diff
@@ -0,0 +1,115 @@
+!(function () {
+  function loadSegment() {
+    var analytics = (window.analytics = window.analytics || []);
+    if (!analytics.initialize)
+      if (analytics.invoked)
+        window.console &&
+          console.error &&
+          console.error('Segment snippet included twice.');
+      else {
+        analytics.invoked = !0;
+        analytics.methods = [
+          'trackSubmit',
+          'trackClick',
+          'trackLink',
+          'trackForm',
+          'pageview',
+          'identify',
+          'reset',
+          'group',
+          'track',
+          'ready',
+          'alias',
+          'debug',
+          'page',
+          'once',
+          'off',
+          'on',
+          'addSourceMiddleware',
+          'addIntegrationMiddleware',
+          'setAnonymousId',
+          'addDestinationMiddleware',
+        ];
+        analytics.factory = function (e) {
+          return function () {
+            var t = Array.prototype.slice.call(arguments);
+            t.unshift(e);
+            analytics.push(t);
+            return analytics;
+          };
+        };
+        for (var e = 0; e < analytics.methods.length; e++) {
+          var key = analytics.methods[e];
+          analytics[key] = analytics.factory(key);
+        }
+        // Fallback callback
+        function n(name) {
+          var g = analytics.factory(name);
+          return function (e, t, n, a) {
+            a && a();
+            return g.apply(null, arguments);
+          };
+        }
+        function x(name) {
+          var g = analytics.factory(name);
+          return function (e, t, n, a, i) {
+            i && i();
+            return g.apply(null, arguments);
+          };
+        }
+        ['identify', 'alias', 'track'].map(function (name) {
+          analytics[name] = n(name);
+        });
+        ['page'].map(function (name) {
+          analytics[name] = x(name);
+        });
+        // Location
+        var req = new XMLHttpRequest();
+        req.open('GET', document.location, false);
+        req.send(null);
+
+        var properties = {
+          City: req.getResponseHeader('x-geoip-city'),
+          Country_code: req.getResponseHeader('x-geoip-country'),
+          Country: req.getResponseHeader('x-geoip-country-name'),
+        };
+
+        function successGetCoordCallback(position) {
+          const coordinates = position.coords;
+          properties = {
+            ...properties,
+            Latitude: coordinates.latitude,
+            Longitude: coordinates.longitude,
+          };
+        }
```

Comment:
Is this part custom? Please mark the start and end along with any other parts since _most_ of the rest of this script is also custom.

Reply:
yes this part is custom. will make it clear